internal/servers/controller/controller.go

```diff
@@ -124,6 +123,25 @@ func New(ctx context.Context, conf *Config) (*Controller, error) {
 		}
 	}
 
+	for _, l := range conf.Listeners {
+		if len(l.Config.Purpose) != 1 {
+			return nil, fmt.Errorf("listener must have exactly one purpose")
+		}
+		switch l.Config.Purpose[0] {
+		case "api":
+			c.apiListeners = append(c.apiListeners, l)
+		case "cluster":
+			c.clusterListeners = append(c.clusterListeners, l)
+		}
+	}
+	if len(c.apiListeners) == 0 {
+		return nil, fmt.Errorf("no api listeners found")
+	}
+	if len(c.clusterListeners) == 0 {
+		return nil, fmt.Errorf("no cluster listeners found")
+	}
+	// TBD: Are we supposed to only allow one cluster listener?
+
```

Comment:
Are we supposed to only allow one cluster listener? Boundary doesn't seem to support multiple cluster listeners.

Reply:
Assuming for now that only one cluster listener is allowed, considering we've seen examples in the codebase that point towards this direction